llama_stack/distribution/utils/memory_bank_utils.py

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+
+from llama_stack.apis.memory_banks.memory_banks import (
+    BankParams,
+    GraphMemoryBank,
+    KeyValueMemoryBank,
+    KeywordMemoryBank,
+    MemoryBank,
+    MemoryBankType,
+    VectorMemoryBank,
+    VectorMemoryBankParams,
+)
+
+
+def build_memory_bank(
+    memory_bank_id: str,
+    params: BankParams,
+    provider_id: str,
+    provider_memorybank_id: str,
+) -> MemoryBank:
+    if params.memory_bank_type == MemoryBankType.vector.value:
```

Comment:
I haven't thought about this deeply but we try to avoid this kind of manual building method by completely relying on pydantic. for example MemoryBank were itself to be an Annotated (tagged) Union, then you could have simply done:
```python
MemoryBank = Annotated[
  Union[VectorMemoryBank, ...],
  Field(discriminator="memory_bank_type")
]

parse_obj_as(MemoryBank, dict(
   memory_bank_id=...,
   provider_id=...,
   **params
))
```

and nothing else would be needed.

Reply:
I think we can have the inheritance while still automating the parsing part. Pushed a new commit with changes. Thoughts?

Reply:
actually, i dont like that path. i am going to get rid of inheritance.